### PR TITLE
[0.27.1rc][Performance][KDA] Drop redundant Kimi KDA padding mask

### PR DESCRIPTION
**Pull request by MaybeChz**
Cherry-pick of PR #15553 onto `releases/v0.27.1rc`.

Original PR: #15553

---
### What this PR does / why we need it?

Kimi KDA previously masked `core_attn_out` after `o_norm` with `query_start_loc[-1]`. That device mask is redundant:

- `num_actual_tokens` is the unpadded live token count from the scheduler.
- Kernel output already overwrites `[:num_actual_tokens]`.
- Dummy seqs with `seqLen=0` live in request-dimension metadata, not extra rows inside the live token slice.

Keep GDN-style `torch.zeros` for the graph window. `o_proj` still runs on `hidden_states.size(0)` (`T_pad`), so `[num_actual_tokens:]` must stay zero instead of `torch.empty` garbage.

Merge still copies the full kernel tensors. Do not Python-slice index buffers to a live count; FULL ACLGraph inlines `_forward` and would bake that length, which is the CANN IndexCheck crash (`1065369473..1065369479`).

### Does this PR introduce _any_ user-facing change?

No. Internal KDA padding handling only.

### How was this patch tested?

- Unit tests in `tests/ut/ops/test_kimi_kda.py`:
  - Reproduces capture-length `index_copy_` against DSpark garbage indices.
  - Decode / spec / mixed `_forward` overwrite live rows.
- Please re-run the Kimi K3 DSpark + ACLGraph accuracy case.

Made with [Cursor](https://cursor.com)

- vLLM main: https://github.com/vllm-project/vllm/commit/ba07e4a48fc951300d97eb506217dd530583dea3

### CODE REVIEW

File: vllm_ascend/ops/kimi_kda.py [hunk 1]
```diff
@@ -164,7 +142,9 @@ def forward(
             g1 = self.f_b_proj(f_a)[0]
             g1 = rearrange(g1, "n (h d) -> 1 n h d", d=self.head_dim)
             g2 = rearrange(g_proj_states, "... (h d) -> ... h d", d=self.head_dim)
-            core_attn_out = torch.empty(
+            # o_proj consumes the full T_pad window. zeros keep
+            # [num_actual_tokens:] from leaking uninitialized values.
+            core_attn_out = torch.zeros(
```
Comment (nu11ptr7) [inline vllm_ascend/ops/kimi_kda.py:248]: With FULL ACLGraph, `forward` runs once at capture, so the `torch.zeros` initialization also happens once - padding rows stay zero only as long as nothing writes them on replay. The index copies now use the full capture length; worth confirming no replay path can write garbage into rows beyond `num_actual_tokens` before o_proj consumes the padded window.
